pkg/server/handler.go

```diff
@@ -30,7 +30,7 @@ import (
 	"k8s.io/kubernetes/pkg/genericcontrolplane"
 )
 
-var reClusterName = regexp.MustCompile(`^[a-z0-9][a-z0-9-]{0,78}[a-z0-9]$`)
+var reClusterName = regexp.MustCompile(`^([a-z0-9][a-z0-9-]{0,78}[a-z0-9]_)?[a-z0-9][a-z0-9-]{0,78}[a-z0-9]$`)
```

Comment:
Would you be able to write a comment describing the format?

Reply:
I don't actually know what the format *is* - it looks like a domain name segment or something similar? I can't tell why there's a length limit, etc - all I did was allow for two of these such things with a separator that's not in the allowed set of characters, to make decoding simple.

Reply:
@smarterclayton @imjasonh @davidfestal any of you know where this regex came from?